ansible/playbooks/etcd-setup.yml

```diff
@@ -0,0 +1,10 @@
+---
```

Comment:
> Why is this at the top level playbooks directory, will each role have a setup play here?

I am following openstack-ansible. Though, I did not absorb every aspect for it.

> Are you envisioning having a deploy-cluster.yml script that would call <role>-setup.yml.

Yes

> To expand we could have a reset-cluster.yml call <role>-reset.yml's In each <role>-reset.yml we use the appropriate tags or includes to enforce the right order and control of tasks executed within that role.

That was the idea. The ``playbooks`` directory would contain ``<role>-setup.yml``, ``<role>-update.yml``, ``<role>-reset.yml`` and others. Each playbook would set appropriate tags (e.g. for ``<role>-update.yml`` it would be ``install`` and ``reset``) and only tagged tasks would be run. However, there is no way to choose tags from within a playbook or role. As based on [1] you can only use ``--tags`` and ``--skip-tags`` flags of the ``ansible-playbook`` command.

> We need to figure out the exactness of how to run something like etcd-setup.yml or an etcd-upgrade.yml. 

Completely agree with that.

> Maybe we just call specific playbooks in the roles themselves tasks with these alternate flows. We can just call the role with an upgrade tag. I think the later would be easiest to do with what ansible provides and what is existing in contrib.

IIUC, you would like to call specific tasks inside a role? By default, ``main.yml`` task is called. Is there a way to call other tasks like ``etcd-start.yml``?

[1] http://stackoverflow.com/questions/25674649/ansible-in-a-playbook-filter-a-role-by-tags-without-passing-at-the-command-l?answertab=votes#tab-top